### DISCUSSION TITLE
4.1.1 breaks worker injection

### DISCUSSION THREAD

**Discussion by Madman600** [ujjwalguptaofficial/JsStore#219] [Q&A]
4.0.6 I had no issues. 4.1.1 with no changes in my code I'm now getting:

connection_helper.ts:43 Uncaught ReferenceError: window is not defined
at Connection.get (connection_helper.ts:43)
at ConnectionHelper.initQueryManager_ (connection_helper.ts:57)
at ConnectionHelper (connection_helper.ts:52)
at new Connection (connection.ts:11)
at DataCache.<instance_members_initializer> (DataCache.js?t=1623074569597:7)
at new DataCache (DataCache.js?t=1623074569597:6)
at DataService.js?t=1623074569597:5

It's throwing the error when I call new JsStore.Connection() before addPlugin is called.

**Comment by ujjwalguptaofficial** [comment]
Hi @Madman600 

I tried reproducing it but couldn't. I have also updated the example with latest version but don't see any issue. Have a look at example - https://github.com/ujjwalguptaofficial/jsstore-examples/tree/master/typescript_without_worker

**Comment by Madman600** [comment]
I was using with vite and vue and for some reason window was undefined but I've changed things to use vite's worker wrapper and gotten it working.

**Reply by Madman600** [reply]
This happens when trying to use jsstore inside a sharedworker. sharedworker can not create worker, and window is undefined.
